[BUG] `npm ping` uses cache when registry is offline

### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

When using `npm ping` to check if a private registry was online, I noticed that it would claim a successful response after approximately 70000ms when the registry was definitely offline.  I'm not sure if it's currently possible to configure the timeout period, but it would be very useful to be able to do so.

The private registry was a [Verdaccio](https://verdaccio.org/) registry hosted on the local machine (launched with `verdaccio --listen 127.0.0.1:4873`), although I'm not sure this is of relevance for this issue.

Suspecting that there might be some sort of caching involved, I ran `npm cache ls`, which confirmed there was an entry in there for the ping request.

Interestingly, after cleaning the entry out of the cache, the next `npm ping` call still succeeds.  It is only the subsequent ones (after the first ping since removing the entry from the cache) which fail as they should.

The commands and their output demonstrating the problem:

### 1) npm ping where the private registry is online:
`npm ping --registry=http://127.0.0.1:4873/ --verbose`
```
npm verb cli C:\Program Files\nodejs\node.exe C:\Users\<REDACTED>\AppData\Roaming\npm\node_modules\npm\bin\npm-cli.js
npm info using npm@9.1.2
npm info using node@v18.12.1
npm verb title npm ping
npm verb argv "ping" "--registry" "http://127.0.0.1:4873/" "--loglevel" "verbose"
npm verb logfile logs-max:10 dir:C:\Users\<REDACTED>\AppData\Local\npm-cache\_logs\2022-11-18T16_05_12_706Z-
npm verb logfile C:\Users\<REDACTED>\AppData\Local\npm-cache\_logs\2022-11-18T16_05_12_706Z-debug-0.log
npm notice PING http://127.0.0.1:4873/
npm http fetch GET 200 http://127.0.0.1:4873/-/ping?write=true 33ms (cache miss)
npm notice PONG 35ms
npm verb exit 0
npm info ok
```

### 2) npm ping where the private registry is offline:
`npm ping --registry=http://127.0.0.1:4873/ --verbose`
```
npm verb cli C:\Program Files\nodejs\node.exe C:\Users\<REDACTED>\AppData\Roaming\npm\node_modules\npm\bin\npm-cli.js
npm info using npm@9.1.2
npm info using node@v18.12.1
npm verb title npm ping
npm verb argv "ping" "--registry" "http://127.0.0.1:4873/" "--loglevel" "verbose"
npm verb logfile logs-max:10 dir:C:\Users\<REDACTED>\AppData\Local\npm-cache\_logs\2022-11-18T16_05_56_431Z-
npm verb logfile C:\Users\<REDACTED>\AppData\Local\npm-cache\_logs\2022-11-18T16_05_56_431Z-debug-0.log
npm notice PING http://127.0.0.1:4873/
npm http fetch GET 200 http://127.0.0.1:4873/-/ping?write=true 70061ms (cache stale)
npm notice PONG 70062ms
npm verb exit 0
npm info ok
```

### 3) npm cache clean the ping entry:
`npm cache clean make-fetch-happen:request-cache:http://127.0.0.1:4873/-/ping?write=true --verbose`
```
npm verb cli C:\Program Files\nodejs\node.exe C:\Users\<REDACTED>\AppData\Roaming\npm\node_modules\npm\bin\npm-cli.js
npm info using npm@9.1.2
npm info using node@v18.12.1
npm verb title npm cache clean make-fetch-happen:request-cache:http://127.0.0.1:4873/-/ping?write=true
npm verb argv "cache" "clean" "make-fetch-happen:request-cache:http://127.0.0.1:4873/-/ping?write=true" "--loglevel" "verbose"
npm verb logfile logs-max:10 dir:C:\Users\<REDACTED>\AppData\Local\npm-cache\_logs\2022-11-18T16_08_04_190Z-
npm verb logfile C:\Users\<REDACTED>\AppData\Local\npm-cache\_logs\2022-11-18T16_08_04_190Z-debug-0.log
Deleted: make-fetch-happen:request-cache:http://127.0.0.1:4873/-/ping?write=true
npm verb exit 0
npm info ok
```

### 4) npm ping where the private registry is offline:
`npm ping --registry=http://127.0.0.1:4873/ --verbose`
```
npm verb cli C:\Program Files\nodejs\node.exe C:\Users\<REDACTED>\AppData\Roaming\npm\node_modules\npm\bin\npm-cli.js
npm info using npm@9.1.2
npm info using node@v18.12.1
npm verb title npm ping
npm verb argv "ping" "--registry" "http://127.0.0.1:4873/" "--loglevel" "verbose"
npm verb logfile logs-max:10 dir:C:\Users\<REDACTED>\AppData\Local\npm-cache\_logs\2022-11-18T16_09_01_885Z-
npm verb logfile C:\Users\<REDACTED>\AppData\Local\npm-cache\_logs\2022-11-18T16_09_01_885Z-debug-0.log
npm notice PING http://127.0.0.1:4873/
npm notice PONG 70076ms
npm verb exit 0
npm info ok
```

### 5) npm ping where the private registry is offline:
`npm ping --registry=http://127.0.0.1:4873/ --verbose`
```
npm ping --registry=http://127.0.0.1:4873/ --verbose
npm verb cli C:\Program Files\nodejs\node.exe C:\Users\<REDACTED>\AppData\Roaming\npm\node_modules\npm\bin\npm-cli.js
npm info using npm@9.1.2
npm info using node@v18.12.1
npm verb title npm ping
npm verb argv "ping" "--registry" "http://127.0.0.1:4873/" "--loglevel" "verbose"
npm verb logfile logs-max:10 dir:C:\Users\<REDACTED>\AppData\Local\npm-cache\_logs\2022-11-18T16_10_51_182Z-
npm verb logfile C:\Users\<REDACTED>\AppData\Local\npm-cache\_logs\2022-11-18T16_10_51_182Z-debug-0.log
npm notice PING http://127.0.0.1:4873/
npm verb type system
npm verb stack FetchError: request to http://127.0.0.1:4873/-/ping?write=true failed, reason: connect ECONNREFUSED 127.0.0.1:4873
npm verb stack     at ClientRequest.<anonymous> (C:\Users\<REDACTED>\AppData\Roaming\npm\node_modules\npm\node_modules\minipass-fetch\lib\index.js:130:14)
npm verb stack     at ClientRequest.emit (node:events:513:28)
npm verb stack     at Socket.socketErrorListener (node:_http_client:494:9)
npm verb stack     at Socket.emit (node:events:525:35)
npm verb stack     at emitErrorNT (node:internal/streams/destroy:151:8)
npm verb stack     at emitErrorCloseNT (node:internal/streams/destroy:116:3)
npm verb stack     at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
npm verb cwd C:\Users\<REDACTED>
npm verb Windows_NT 10.0.22621
npm verb node v18.12.1
npm verb npm  v9.1.2
npm ERR! code ECONNREFUSED
npm ERR! syscall connect
npm ERR! errno ECONNREFUSED
npm ERR! FetchError: request to http://127.0.0.1:4873/-/ping?write=true failed, reason: connect ECONNREFUSED 127.0.0.1:4873
npm ERR!     at ClientRequest.<anonymous> (C:\Users\<REDACTED>\AppData\Roaming\npm\node_modules\npm\node_modules\minipass-fetch\lib\index.js:130:14)
npm ERR!     at ClientRequest.emit (node:events:513:28)
npm ERR!     at Socket.socketErrorListener (node:_http_client:494:9)
npm ERR!     at Socket.emit (node:events:525:35)
npm ERR!     at emitErrorNT (node:internal/streams/destroy:151:8)
npm ERR!     at emitErrorCloseNT (node:internal/streams/destroy:116:3)
npm ERR!     at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
npm ERR!  FetchError: request to http://127.0.0.1:4873/-/ping?write=true failed, reason: connect ECONNREFUSED 127.0.0.1:4873
npm ERR!     at ClientRequest.<anonymous> (C:\Users\<REDACTED>\AppData\Roaming\npm\node_modules\npm\node_modules\minipass-fetch\lib\index.js:130:14)
npm ERR!     at ClientRequest.emit (node:events:513:28)
npm ERR!     at Socket.socketErrorListener (node:_http_client:494:9)
npm ERR!     at Socket.emit (node:events:525:35)
npm ERR!     at emitErrorNT (node:internal/streams/destroy:151:8)
npm ERR!     at emitErrorCloseNT (node:internal/streams/destroy:116:3)
npm ERR!     at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
npm ERR!   code: 'ECONNREFUSED',
npm ERR!   errno: 'ECONNREFUSED',
npm ERR!   syscall: 'connect',
npm ERR!   address: '127.0.0.1',
npm ERR!   port: 4873,
npm ERR!   type: 'system'
npm ERR! }
npm ERR!
npm ERR! If you are behind a proxy, please make sure that the
npm ERR! 'proxy' config is set properly.  See: 'npm help config'
npm verb exit 1
npm verb code 1

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\<REDACTED>\AppData\Local\npm-cache\_logs\2022-11-18T16_10_51_182Z-debug-0.log
```

### Expected Behavior

`npm ping` responses should not be cached.  It should accurately report whether or not the specified registry is actually online, rather than returning a cached response after timing out.

### Steps To Reproduce

1. With the private registry **online**, run the following command: `npm ping --registry=http://127.0.0.1:4873/ --verbose`
2. Shut down the private registry.
3. With the private registry **offline**, run the following command: `npm ping --registry=http://127.0.0.1:4873/ --verbose`

### Environment

- **npm**: 9.1.2
- **Node.js**: 18.12.1
- **OS Name**: Windows 11 22H2 (OS Build 22621.674)
- **System Model Name**: N/A
- **npm config**:
```
; "builtin" config from C:\Users\<REDACTED>\AppData\Roaming\npm\node_modules\npm\npmrc

prefix = "C:\\Users\\<REDACTED>\\AppData\\Roaming\\npm"

; "cli" config from command line options

location = "project"

; node bin location = C:\Program Files\nodejs\node.exe
; node version = v18.12.1
; npm local prefix = C:\Users\<REDACTED>
; npm version = 9.1.2
; cwd = C:\Users\<REDACTED>
; HOME = C:\Users\<REDACTED>
; Run `npm config ls -l` to show all defaults.
```

--- a/lib/utils/ping.js
+++ b/lib/utils/ping.js
@@ -2,6 +2,6 @@
 // used by the ping and doctor commands
 const fetch = require('npm-registry-fetch')
 module.exports = async (flatOptions) => {
-  const res = await fetch('/-/ping?write=true', flatOptions)
+  const res = await fetch('/-/ping', { ...flatOptions, cache: false })
   return res.json().catch(() => ({}))
 }

--- a/mock-registry/lib/index.js
+++ b/mock-registry/lib/index.js
@@ -320,7 +320,7 @@ class MockRegistry {
   }
 
   ping ({ body = {}, responseCode = 200 } = {}) {
-    this.nock = this.nock.get(this.fullPath('/-/ping?write=true')).reply(responseCode, body)
+    this.nock = this.nock.get(this.fullPath('/-/ping')).reply(responseCode, body)
   }
 
   // full unpublish of an entire package

--- a/tap-snapshots/test/lib/commands/doctor.js.test.cjs
+++ b/tap-snapshots/test/lib/commands/doctor.js.test.cjs
@@ -1166,7 +1166,7 @@ Object {
 exports[`test/lib/commands/doctor.js TAP ping 404 > ping 404 1`] = `
 Connecting to the registry
 Not ok
-404 404 Not Found - GET https://registry.npmjs.org/-/ping?write=true
+404 404 Not Found - GET https://registry.npmjs.org/-/ping
 Checking npm version
 Ok
 current: v1.0.0, latest: v1.0.0
@@ -1226,7 +1226,7 @@ Object {
 exports[`test/lib/commands/doctor.js TAP ping 404 in color > ping 404 in color 1`] = `
 Connecting to the registry
 [31mNot ok[39m
-[36m404 404 Not Found - GET https://registry.npmjs.org/-/ping?write=true[39m
+[36m404 404 Not Found - GET https://registry.npmjs.org/-/ping[39m
 Checking npm version
 [32mOk[39m
 current: v1.0.0, latest: v1.0.0
@@ -1286,7 +1286,7 @@ Object {
 exports[`test/lib/commands/doctor.js TAP ping exception with code > ping failure 1`] = `
 Connecting to the registry
 Not ok
-request to https://registry.npmjs.org/-/ping?write=true failed, reason: Test Error
+request to https://registry.npmjs.org/-/ping failed, reason: Test Error
 Checking npm version
 Ok
 current: v1.0.0, latest: v1.0.0
@@ -1346,7 +1346,7 @@ Object {
 exports[`test/lib/commands/doctor.js TAP ping exception without code > ping failure 1`] = `
 Connecting to the registry
 Not ok
-request to https://registry.npmjs.org/-/ping?write=true failed, reason: Test Error
+request to https://registry.npmjs.org/-/ping failed, reason: Test Error
 Checking npm version
 Ok
 current: v1.0.0, latest: v1.0.0

--- a/test/lib/commands/doctor.js
+++ b/test/lib/commands/doctor.js
@@ -89,7 +89,7 @@ t.test('all clear', async t => {
     ...dirs,
   })
   tnock(t, npm.config.get('registry'))
-    .get('/-/ping?write=true').reply(200, '{}')
+    .get('/-/ping').reply(200, '{}')
     .get('/npm').reply(200, npmManifest(npm.version))
   tnock(t, 'https://nodejs.org')
     .get('/dist/index.json').reply(200, nodeVersions)
@@ -108,7 +108,7 @@ t.test('all clear in color', async t => {
     },
   })
   tnock(t, npm.config.get('registry'))
-    .get('/-/ping?write=true').reply(200, '{}')
+    .get('/-/ping').reply(200, '{}')
     .get('/npm').reply(200, npmManifest(npm.version))
   tnock(t, 'https://nodejs.org')
     .get('/dist/index.json').reply(200, nodeVersions)
@@ -127,7 +127,7 @@ t.test('silent success', async t => {
     ...dirs,
   })
   tnock(t, npm.config.get('registry'))
-    .get('/-/ping?write=true').reply(200, '{}')
+    .get('/-/ping').reply(200, '{}')
     .get('/npm').reply(200, npmManifest(npm.version))
   tnock(t, 'https://nodejs.org')
     .get('/dist/index.json').reply(200, nodeVersions)
@@ -146,7 +146,7 @@ t.test('silent errors', async t => {
     ...dirs,
   })
   tnock(t, npm.config.get('registry'))
-    .get('/-/ping?write=true').reply(404, '{}')
+    .get('/-/ping').reply(404, '{}')
   await t.rejects(npm.exec('doctor', ['ping']), {
     message: /Check logs/,
   })
@@ -161,7 +161,7 @@ t.test('ping 404', async t => {
     ...dirs,
   })
   tnock(t, npm.config.get('registry'))
-    .get('/-/ping?write=true').reply(404, '{}')
+    .get('/-/ping').reply(404, '{}')
     .get('/npm').reply(200, npmManifest(npm.version))
   tnock(t, 'https://nodejs.org')
     .get('/dist/index.json').reply(200, nodeVersions)
@@ -182,7 +182,7 @@ t.test('ping 404 in color', async t => {
     },
   })
   tnock(t, npm.config.get('registry'))
-    .get('/-/ping?write=true').reply(404, '{}')
+    .get('/-/ping').reply(404, '{}')
     .get('/npm').reply(200, npmManifest(npm.version))
   tnock(t, 'https://nodejs.org')
     .get('/dist/index.json').reply(200, nodeVersions)
@@ -198,7 +198,7 @@ t.test('ping exception with code', async t => {
     ...dirs,
   })
   tnock(t, npm.config.get('registry'))
-    .get('/-/ping?write=true').replyWithError({ message: 'Test Error', code: 'TEST' })
+    .get('/-/ping').replyWithError({ message: 'Test Error', code: 'TEST' })
     .get('/npm').reply(200, npmManifest(npm.version))
   tnock(t, 'https://nodejs.org')
     .get('/dist/index.json').reply(200, nodeVersions)
@@ -214,7 +214,7 @@ t.test('ping exception without code', async t => {
     ...dirs,
   })
   tnock(t, npm.config.get('registry'))
-    .get('/-/ping?write=true').replyWithError({ message: 'Test Error', code: false })
+    .get('/-/ping').replyWithError({ message: 'Test Error', code: false })
     .get('/npm').reply(200, npmManifest(npm.version))
   tnock(t, 'https://nodejs.org')
     .get('/dist/index.json').reply(200, nodeVersions)
@@ -230,7 +230,7 @@ t.test('npm out of date', async t => {
     ...dirs,
   })
   tnock(t, npm.config.get('registry'))
-    .get('/-/ping?write=true').reply(200, '{}')
+    .get('/-/ping').reply(200, '{}')
     .get('/npm').reply(200, npmManifest('2.0.0'))
   tnock(t, 'https://nodejs.org')
     .get('/dist/index.json').reply(200, nodeVersions)
@@ -255,7 +255,7 @@ t.test('node out of date - lts', async t => {
     ...dirs,
   })
   tnock(t, npm.config.get('registry'))
-    .get('/-/ping?write=true').reply(200, '{}')
+    .get('/-/ping').reply(200, '{}')
     .get('/npm').reply(200, npmManifest(npm.version))
   tnock(t, 'https://nodejs.org')
     .get('/dist/index.json').reply(200, nodeVersions)
@@ -280,7 +280,7 @@ t.test('node out of date - current', async t => {
     ...dirs,
   })
   tnock(t, npm.config.get('registry'))
-    .get('/-/ping?write=true').reply(200, '{}')
+    .get('/-/ping').reply(200, '{}')
     .get('/npm').reply(200, npmManifest(npm.version))
   tnock(t, 'https://nodejs.org')
     .get('/dist/index.json').reply(200, nodeVersions)
@@ -297,7 +297,7 @@ t.test('non-default registry', async t => {
     ...dirs,
   })
   tnock(t, npm.config.get('registry'))
-    .get('/-/ping?write=true').reply(200, '{}')
+    .get('/-/ping').reply(200, '{}')
     .get('/npm').reply(200, npmManifest(npm.version))
   tnock(t, 'https://nodejs.org')
     .get('/dist/index.json').reply(200, nodeVersions)
@@ -318,7 +318,7 @@ t.test('missing git', async t => {
     ...dirs,
   })
   tnock(t, npm.config.get('registry'))
-    .get('/-/ping?write=true').reply(200, '{}')
+    .get('/-/ping').reply(200, '{}')
     .get('/npm').reply(200, npmManifest(npm.version))
   tnock(t, 'https://nodejs.org')
     .get('/dist/index.json').reply(200, nodeVersions)
@@ -344,7 +344,7 @@ t.test('windows skips permissions checks', async t => {
     globalPrefixDir: {},
   })
   tnock(t, npm.config.get('registry'))
-    .get('/-/ping?write=true').reply(200, '{}')
+    .get('/-/ping').reply(200, '{}')
     .get('/npm').reply(200, npmManifest(npm.version))
   tnock(t, 'https://nodejs.org')
     .get('/dist/index.json').reply(200, nodeVersions)
@@ -361,7 +361,7 @@ t.test('missing global directories', async t => {
     globalPrefixDir: {},
   })
   tnock(t, npm.config.get('registry'))
-    .get('/-/ping?write=true').reply(200, '{}')
+    .get('/-/ping').reply(200, '{}')
     .get('/npm').reply(200, npmManifest(npm.version))
   tnock(t, 'https://nodejs.org')
     .get('/dist/index.json').reply(200, nodeVersions)
@@ -377,7 +377,7 @@ t.test('missing local node_modules', async t => {
     globalPrefixDir: dirs.globalPrefixDir,
   })
   tnock(t, npm.config.get('registry'))
-    .get('/-/ping?write=true').reply(200, '{}')
+    .get('/-/ping').reply(200, '{}')
     .get('/npm').reply(200, npmManifest(npm.version))
   tnock(t, 'https://nodejs.org')
     .get('/dist/index.json').reply(200, nodeVersions)
@@ -406,7 +406,7 @@ t.test('incorrect owner', async t => {
     ...dirs,
   })
   tnock(t, npm.config.get('registry'))
-    .get('/-/ping?write=true').reply(200, '{}')
+    .get('/-/ping').reply(200, '{}')
     .get('/npm').reply(200, npmManifest(npm.version))
   tnock(t, 'https://nodejs.org')
     .get('/dist/index.json').reply(200, nodeVersions)
@@ -430,7 +430,7 @@ t.test('incorrect permissions', async t => {
     ...dirs,
   })
   tnock(t, npm.config.get('registry'))
-    .get('/-/ping?write=true').reply(200, '{}')
+    .get('/-/ping').reply(200, '{}')
     .get('/npm').reply(200, npmManifest(npm.version))
   tnock(t, 'https://nodejs.org')
     .get('/dist/index.json').reply(200, nodeVersions)
@@ -458,7 +458,7 @@ t.test('error reading directory', async t => {
     ...dirs,
   })
   tnock(t, npm.config.get('registry'))
-    .get('/-/ping?write=true').reply(200, '{}')
+    .get('/-/ping').reply(200, '{}')
     .get('/npm').reply(200, npmManifest(npm.version))
   tnock(t, 'https://nodejs.org')
     .get('/dist/index.json').reply(200, nodeVersions)
@@ -481,7 +481,7 @@ t.test('cacache badContent', async t => {
     ...dirs,
   })
   tnock(t, npm.config.get('registry'))
-    .get('/-/ping?write=true').reply(200, '{}')
+    .get('/-/ping').reply(200, '{}')
     .get('/npm').reply(200, npmManifest(npm.version))
   tnock(t, 'https://nodejs.org')
     .get('/dist/index.json').reply(200, nodeVersions)
@@ -504,7 +504,7 @@ t.test('cacache reclaimedCount', async t => {
     ...dirs,
   })
   tnock(t, npm.config.get('registry'))
-    .get('/-/ping?write=true').reply(200, '{}')
+    .get('/-/ping').reply(200, '{}')
     .get('/npm').reply(200, npmManifest(npm.version))
   tnock(t, 'https://nodejs.org')
     .get('/dist/index.json').reply(200, nodeVersions)
@@ -527,7 +527,7 @@ t.test('cacache missingContent', async t => {
     ...dirs,
   })
   tnock(t, npm.config.get('registry'))
-    .get('/-/ping?write=true').reply(200, '{}')
+    .get('/-/ping').reply(200, '{}')
     .get('/npm').reply(200, npmManifest(npm.version))
   tnock(t, 'https://nodejs.org')
     .get('/dist/index.json').reply(200, nodeVersions)
@@ -558,7 +558,7 @@ t.test('discrete checks', t => {
       ...dirs,
     })
     tnock(t, npm.config.get('registry'))
-      .get('/-/ping?write=true').reply(200, '{}')
+      .get('/-/ping').reply(200, '{}')
     await npm.exec('doctor', ['ping'])
     t.matchSnapshot(joinedOutput(), 'output')
     t.matchSnapshot({ info: logs.info, warn: logs.warn, error: logs.error }, 'logs')
@@ -586,7 +586,7 @@ t.test('discrete checks', t => {
       ...dirs,
     })
     tnock(t, npm.config.get('registry'))
-      .get('/-/ping?write=true').reply(200, '{}')
+      .get('/-/ping').reply(200, '{}')
     await npm.exec('doctor', ['registry'])
     t.matchSnapshot(joinedOutput(), 'output')
     t.matchSnapshot({ info: logs.info, warn: logs.warn, error: logs.error }, 'logs')

--- a/test/lib/commands/ping.js
+++ b/test/lib/commands/ping.js
@@ -1,6 +1,8 @@
 const t = require('tap')
 const { load: loadMockNpm } = require('../../fixtures/mock-npm.js')
 const MockRegistry = require('@npmcli/mock-registry')
+const cacache = require('cacache')
+const path = require('node:path')
 
 t.test('no details', async t => {
   const { npm, logs, joinedOutput } = await loadMockNpm(t)
@@ -73,4 +75,20 @@ t.test('invalid json', async t => {
     time: /[0-9]+/,
     details: {},
   })
+})
+t.test('fail when registry is unreachable even if request is cached', async t => {
+  const { npm } = await loadMockNpm(t, {
+    config: { registry: 'https://ur.npmlocal.npmtest/' },
+    cacheDir: { _cacache: {} },
+  })
+  const url = `${npm.config.get('registry')}-/ping`
+  const cache = path.join(npm.cache, '_cacache')
+  await cacache.put(cache,
+    `make-fetch-happen:request-cache:${url}`,
+    '{}', { metadata: { url } }
+  )
+  t.rejects(npm.exec('ping', []), {
+    code: 'ENOTFOUND',
+  },
+  'throws ENOTFOUND error')
 })

